internal/reconciliations/istio/installation.go

```diff
@@ -82,6 +85,11 @@ func (i *Installation) Reconcile(ctx context.Context, istioCR operatorv1alpha1.I
 			return istioCR, described_errors.NewDescribedError(err, "Could not get configuration from Istio Operator file")
 		}
 
+		ingressGatewayNeedsRestart, err := ingressgateway.NeedsRestart(ctx, i.Client, &istioCR)
```

Comment:
👍 Good job, this should work. The only downside I can see is, that we don't rely on the real state of the Istio resources, but our assumption that is stored as last applied config in the Istio CR. But since we can't easily find out what's the current configuration of the IGWs, it's better to use our last applied annotation than Istios config map. Still a user could modify it and we'd not be able to detect it, but I'd consider this an absolut corner case.